Code/src/main/java/nl/utwente/group10/haskell/type/ConstT.java

```diff
@@ -62,6 +67,30 @@ public String toHaskellType() {
     }
 
     @Override
+    public ConstT getFresh() {
+        return new ConstT(this.constructor, this.getFreshArgs());
+    }
+
+    /**
+     * Returns an array of fresh arguments. Selectively calls {@code getFresh} on each argument. When the same Type
+     * instance appears multiple times in the arguments no new type is instantiated. Instead, the fresh type is reused.
+     * @return An array of fresh arguments.
+     */
+    protected Type[] getFreshArgs() {
+        List<Type> fresh = new LinkedList<>();
```

Comment:
Me too. I might have written this in a partially awake state. However, a linked list might be efficient as there usually are one or two arguments. (This is not the case with tuples, but since we are only supporting 2-tuples, it might ultimately be true for tuples as well.)